torch/distributed/_tensor/placement_types.py

```diff
@@ -308,15 +316,27 @@ def _replicate_tensor(
         """
         Replicate (broadcast) a torch.Tensor on a mesh dimension (use
         the first coordinate on the mesh dimension as source of truth)
+
+        if tensor is on meta, then just allocate an empty tensor on the desired device based on the device mesh's
+        device_type
         """
         my_coordinate = mesh.get_coordinate()
         if my_coordinate is None:
             # if rank is not part of mesh, we simply return an empty tensor
-            return tensor.new_empty(0, requires_grad=tensor.requires_grad)
+            return tensor.new_empty(
+                0, device=mesh.device_type, requires_grad=tensor.requires_grad
+            )
 
         tensor = tensor.contiguous()
-        mesh.broadcast(tensor, mesh_dim=mesh_dim)
-        return tensor
+        if tensor.is_meta:
+            return torch.empty_like(tensor, device=mesh.device_type)
```

Comment:
same here, I think we should only do `to_empty` inside `distribute_module` call, and don't eagerly initialize the memory inside `distribute_tensor` (i.e. _shard_tensor/_replicate_tensor)

Reply:
I've removed it